x-pack/plugins/apm/server/routes/alerts/rule_types/utils/get_groupby_terms.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ENVIRONMENT_NOT_DEFINED } from '../../../../../common/environment_filter_values';
+import { SERVICE_ENVIRONMENT } from '../../../../../common/es_fields/apm';
+
+export const getGroupByTerms = (groupBy: string[] | undefined) => {
+  return (groupBy ?? []).map((group) => {
+    return {
+      field: group,
+      missing:
+        group === SERVICE_ENVIRONMENT
+          ? ENVIRONMENT_NOT_DEFINED.value
+          : group.replaceAll('.', '_').toUpperCase().concat('_NOT_DEFINED'),
+    };
+  });
+};
```

Comment:
The "Not defined" behaviour is specifically for `service.environment`. It is technical debt and ideally we want to remove it in the future. We do not want to apply this behaviour to other fields.

```suggestion
export const getGroupByTerms = (groupByFields: string[] | undefined = []) => {
  return groupByFields.map((groupByField) => {
    return {
      field: groupByField,
      missing:
        groupByField === SERVICE_ENVIRONMENT
          ? ENVIRONMENT_NOT_DEFINED.value
          : undefined,
    };
  });
};
```

Btw I updated the naming to align with the caller (`groupByFields` instead of `groupBy`) abd added a default param so we don't have to handle that in the function body.

Reply:
If we remove `missing: ENVIRONMENT_NOT_DEFINED` from the `multi_terms` aggregation, and if the `environment` field is not there, will there be still a bucket key for `environment`? This will need further testing.

I wasn't aware this was a technical debt. I think we should address all technical debt related to fields being not defined in a separate PR. What do you think?

Reply:
> If we remove `missing: ENVIRONMENT_NOT_DEFINED` from the multi_terms aggregation

We should not remove `missing: ENVIRONMENT_NOT_DEFINED` when field is `SERVICE_ENVIRONMENT`. On the contrary: we should **only** have a missing bucket for environment.

Reply:
I saw the missing bucket for transaction name too when testing on CCS. What is the disadvantage of keeping NOT_DEFINED for every field in aggregation?